scripts/ci/check_colors.py

```diff
@@ -0,0 +1,141 @@
+
+import argparse
+import os
+import re
+
+# import xml.etree.ElementTree as etree
+
+HEX_COLOR_REGEX = r"#[0-9a-fA-F]{6}"
+COLOR_DEF_REGEX = r"color\..* ="
+COLOR_USE_REGEX = r"MZTheme\.colors\.[0-9A-Za-z]+"
+
+def fileContents(filepath):
+    try:
+        with open(filepath, 'r') as file:
+            content = file.read()
+            if not content:
+                exit(f"No content found in: {filepath}")        
+            return content
+    except FileNotFoundError:
+        exit(f"File not found: {filepath}")
+    except Exception as e:
+        exit(f"An error occurred while loading {filepath}: {e}")
+
+def checkForHexColor(filepath):
+    # print("Checking for hex colors in " + filepath)
+    content = fileContents(filepath)
+    hex_color_list = re.findall(HEX_COLOR_REGEX, content)
+    if hex_color_list:
+        exit(f"Unexpected hex color found in: {filepath}")
+
+def colorsDefinedInFile(filepath):
+    print("Loading colors defined in " + filepath)
+    content = fileContents(filepath)
+    color_list = re.findall(COLOR_DEF_REGEX, content)
+    if not color_list:
+        exit(f"No colors found in: {filepath}")
+    # Remove the " =" at the end of each line
+    color_list = list(map(lambda x: x.split(" ")[0], color_list))
+    # Remove the "color." at the start
+    color_list = list(map(lambda x: x.split(".")[1], color_list))
+    return color_list
+
+def colorsUsedInFile(filepath):
+    # print("Loading colors used in " + filepath)
+    content = fileContents(filepath)
+    color_list = re.findall(COLOR_USE_REGEX, content)
+    # Remove the "MZTheme.colors." at the start of each line
+    color_list = list(map(lambda x: x.split(".")[2], color_list))
+    return color_list
+
+def getQmlFiles(directory):
+    print("Loading all QML files in " + directory)
+    qml_files = []
+    try:
+        for root, _, files in os.walk(directory):
+            for file in files:
+                if file.endswith(".qml"):
+                    qml_files.append(os.path.join(root, file))
+        if not qml_files:
+            exit(f"No QML files found in: {directory}")
+        return qml_files
+    except FileNotFoundError:
+        exit(f"Directory not found: {directory}")
+    except Exception as e:
+        exit(f"An error occurred when getting QML files in {directory}: {e}")
+
+parser = argparse.ArgumentParser()
+parser.add_argument('themeDirectory', metavar='themeDirectory', type=str, nargs=1,
+                    help='The directory for all theming')
+args = parser.parse_args()
+
+###
+# 0. Get list of all themes
+try:
+    all_themes = [item for item in os.listdir(args.themeDirectory[0]) if os.path.isdir(os.path.join(args.themeDirectory[0], item))]
+except FileNotFoundError:
+    exit(f"Path not found: {args.themeDirectory[0]}")
+except Exception as e:
+    exit(f"An error occurred when finding all themes: {e}")
+
+if len(all_themes) == 0:
+    exit(f"No themes found")
+
+###
+# 1. All theme's `theme.js` should have the same colors
```

Comment:
Minor nit - if we intend on enforcing this as a lint, then we should just have a single copy of `colors.js` that gets used by all themes (eg: by class inheritance or file linking or something).

Reply:
To make sure I understand, which information are you looking for a single copy of? Currently, there is only a single copy of `colors.js`.

There are 3 files assembled togethered for colors:
- `colors.js`- These are the base colors, including hex codes. There is only one copy of it, shared among all themes.
- `theme.js` - This is the custom work done for each theme. It can only refer to colors in `colors.js`, and all `theme.js` files need to expose the exact same named colors.
- `theme-derived.js` - some objects that are assembled from colors picked in `theme.js`. There is only one copy of it, and it is shared among all themes.

The colors in `theme.js` and `theme-derived.js` are the only ones that can be used in QML files.

Reply:
Oh, okay, so the idea here is that we are just checking that every `theme.js` defines the same set of colors but the values of those colors can be different for each theme...  and every theme can pull from a common `colors.js` for the hex codes.

Please disregard my nit then - this is good as is.